gatling-jms/src/main/scala/io/gatling/jms/check/JmsMessageBodyPreparers.scala

```diff
@@ -16,25 +16,29 @@
 
 package io.gatling.jms.check
 
+import io.gatling.commons.validation._
 import io.gatling.core.check.Preparer
+import io.gatling.core.config.GatlingConfiguration
 import javax.jms.{ BytesMessage, Message, TextMessage }
-import io.gatling.commons.validation._
 
 object JmsMessageBodyPreparers {
 
-  val JmsStringBodyPreparer: Preparer[Message, String] = {
+  protected[check] def toBytes(bytesMessage: BytesMessage): Array[Byte] = {
+    val buffer = Array.ofDim[Byte](bytesMessage.getBodyLength.toInt)
+    bytesMessage.readBytes(buffer)
+    buffer
+  }
+
+  def JmsStringBodyPreparer(implicit config: GatlingConfiguration): Preparer[Message, String] = {
     case tm: TextMessage  => tm.getText.success
-    case bm: BytesMessage => bm.readUTF.success
+    case bm: BytesMessage => new String(toBytes(bm), config.core.charset).success
```

Comment:
is readUTF implemented in an optimized way that it would be worth adding a short path if charset is UTF-8?